pkg/event-consumers/kinesis/kinesis-consumer.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright (c) 2016-2017 Bitnami
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kinesis
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	kubelessApi "github.com/kubeless/kubeless/pkg/apis/kubeless/v1beta1"
+	"github.com/kubeless/kubeless/pkg/utils"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	mutex            = &sync.Mutex{}
+	stopM            map[string](chan struct{})
+	stoppedM         map[string](chan struct{})
+	streamProcessors map[string]bool
+	url              string
+)
+
+func init() {
+	stopM = make(map[string](chan struct{}))
+	stoppedM = make(map[string](chan struct{}))
+	streamProcessors = make(map[string]bool)
+}
+
+// createStreamProcessor polls and gets messages from given AWS kinesis stream and send the stream records to function service
+func createStreamProcessor(triggerObj *kubelessApi.KinesisTrigger, funcName, ns string, clientset kubernetes.Interface, stopchan, stoppedchan chan struct{}) {
+
+	// TODO: for now use 1 sec as polling period to poll Kinesis stream. But need to figure right value as per best practice, as let user specify
```

Comment:
why don't you apply this TODO now? should be easy to add a `poll interval` parameter into the CRD defaulting it to 1s

Reply:
I cross checked with Lambda + Kinesis documntaion there is explicit poll period that user can specify. But Lambda service does polling every `250ms`. I will hard code that value for now.